internal/db/rules.go

```diff
@@ -1,45 +1,175 @@
 package db
 
 import (
-	"database/sql"
 	"fmt"
 
+	"log"
+	"strconv"
+
+	"encoding/json"
+
 	// Blank import required for mysql driver
 	_ "github.com/go-sql-driver/mysql"
 	"github.com/jimmyjames85/bouncecm/internal/models"
 )
 
 // ListRules - Function to pull all rules from db
-func ListRules() (models.RulesObject, error) {
+func (c *Client) ListRules() (*models.RulesObject, error) {
 	rules := []models.BounceRule{}
 
-	db, err := sql.Open("mysql", "root:root@tcp(127.0.0.1:3306)/drop_rules")
+	rows, err := c.Conn.Query("SELECT * FROM bounce_rule")
 
-	checkErr(err)
+	if err != nil {
+		log.Println(err)
+		fmt.Println("FAILEDHERE?")
 
-	rows, err := db.Query("SELECT * FROM bounce_rule")
+		return nil, err;
 
-	checkErr(err)
+	}
 
 	for rows.Next() {
 		br := models.BounceRule{}
+		var description *string
+		err := rows.Scan(&br.ID, &br.ResponseCode, &br.EnhancedCode, &br.Regex, &br.Priority, &description , &br.BounceAction)
+		
+		// case when description is null becuase it can be null according to droprules.sql
+		if description == nil {
+			br.Description = ""
+		} else {
+			br.Description = *description
+		}
+
+
+		if err != nil {
+			log.Println(err)
+			fmt.Println(br)
+
+			return nil, err;
+		}
 
-		err = rows.Scan(&br.ID, &br.ResponseCode, &br.EnhancedCode, &br.Regex, &br.Priority, &br.Description, &br.BounceAction)
-		checkErr(err)
 		rules = append(rules, br)
+		
 	}
 
-	defer rows.Close()
+	rulesObject := models.RulesObject{Rules: rules, NumRules: len(rules)}
 
-	db.Close()
+	return &rulesObject, nil
+}
 
-	rulesObject := models.RulesObject{Rules: rules, NumRules: len(rules)}
+func (c *Client) CreateRuleDB(rule *models.BounceRule) ([]byte, error) {
+	stmt, err := c.Conn.Prepare("INSERT INTO bounce_rule(response_code,enhanced_code,regex,priority,description,bounce_action) VALUES(?,?,?,?,?,?)")
+	
+	if err != nil {
+		log.Println(err)
+		return nil, err;
+	}
+
+
+	_, err = stmt.Exec(rule.ResponseCode, rule.EnhancedCode, rule.Regex, rule.Priority, rule.Description, rule.BounceAction)
+	
+	if err != nil {
+		log.Println(err)
+		return nil, err;
+	}
 
-	return rulesObject, nil
+	data, err := json.Marshal(rule)
+	
+	if err != nil {
+		log.Println(err)
+		return nil, err;
+	}
+	return data, nil
 }
 
-func checkErr(err error) {
+func (c *Client) GetRuleDB(id string) (*models.BounceRule, error) {
+	var bounce_rule *models.BounceRule
+	rows, err := c.Conn.Query("SELECT * FROM bounce_rule WHERE id=" + id)
+	for rows.Next() {
+		var br models.BounceRule
+		err = rows.Scan(&br.ID, &br.ResponseCode, &br.EnhancedCode, &br.Regex, &br.Priority, &br.Description, &br.BounceAction)
+		if err != nil {
+			log.Println(err)
+			return nil, err;
+		}
+		bounce_rule = &br
+		return bounce_rule, nil
+	}
+	return nil, err
+}
+
+func (c *Client) DeleteRuleDB(id int) (error){
+	query := fmt.Sprintf("%s%d", "DELETE FROM bounce_rule WHERE id=", id)
+	_, err := c.Conn.Query(query)
```

Comment:
should be an `Exec` since we are not actually querying for information. Here is a [good read](http://jamestappe.com/The_Ultimate_Guide_To_Building_Database-Driven_Apps_with_Go.pdf) by VividCortex. It explains when to use which function, and what happens under the hood to the connection pool.